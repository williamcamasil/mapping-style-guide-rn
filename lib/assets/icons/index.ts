export * as Large from './large';
export * as Heavy from './heavy';
export * as Thin from './thin';
export * as Small from './small';
export * as Default from './default';
