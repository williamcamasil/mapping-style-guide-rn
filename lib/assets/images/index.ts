export { default as BtnTakePicture } from './BtnTakePicture';
export { default as BulletPoint } from './BulletPoint';
export { default as CheckboxImg } from './CheckboxImg';
export { default as DialogClose } from './DialogClose/index.svg';
export { default as Lock } from './Lock';
export { default as MaskedCardNumber } from './MaskedCardNumber';
export { default as Pointer } from './Pointer';
export { default as RadioImg } from './RadioImg';
