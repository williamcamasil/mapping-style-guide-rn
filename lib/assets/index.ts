export * as Icons from './icons';
export * as Images from './images';
export * as Lotties from './lotties';
