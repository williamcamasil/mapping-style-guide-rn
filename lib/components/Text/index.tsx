import React from 'react';
import {
  StyleProp, StyleSheet, Text as RNText, TextProps as RNTextProp, TextStyle,
} from 'react-native';

import { useTextStyles } from '../../hooks';
import { withTheme } from '../../theme';
import { AppThemeType } from '../../theme/default';
import {
  ColorPaletesType, FontWeightType, LineHeightType, Typography,
} from '../../tokens';
import {
  getColorStyle, getVariantStyle, getWeightStyle, TextVariantType,
} from './utils';

export type { TextVariantType } from './utils';

export type TextPropTypes = RNTextProp & {
  theme: AppThemeType;
  style?: StyleProp<TextStyle>;
  color?: ColorPaletesType;
  variant?: TextVariantType;
  weight?: FontWeightType;
  lineHeight?: LineHeightType;
};

const styles = StyleSheet.create({
  text: {
    letterSpacing: Typography.letterSpaces.default,
    flexShrink: 1,
  },
});

const Text: React.FC<TextPropTypes> = ({
  children,
  color,
  theme,
  style,
  variant = 'body',
  weight,
  lineHeight,
  allowFontScaling = false,
  ...others
}) => {

  const textStyles = useTextStyles(() => [
    styles.text,
    getColorStyle(theme, color ?? 'neutralGray600'),
    getVariantStyle(theme, variant, lineHeight),
    getWeightStyle(theme, weight),
    style,
  ], [color, lineHeight, style, theme, variant, weight]);

  return (
    <RNText
      {...others}
      allowFontScaling={allowFontScaling}
      style={textStyles}
    >
      {children}
    </RNText>
  );
};

export default withTheme(Text);
