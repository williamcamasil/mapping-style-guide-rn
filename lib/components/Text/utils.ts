import { TextStyle } from 'react-native';

import { AppThemeType } from '../../theme/default';
import { ColorPaletesType, FontWeightType, LineHeightType } from '../../tokens';
import { calcLineHeight } from '../../utils/typography';

export type TextVariantType =
  // heading
  'headingDisplay' |
  'headingXLarge' |
  'headingLarge' |
  'headingMedium' |
  'headingSmall' |
  // paragraph regular
  'subtitle' |
  'body' |
  'bodySmall' |
  'caption' |
  // small
  'small' |
  'smallCaps' |
  // one line
  'oneLineHuge' |
  'oneLineLarge' |
  'oneLineMedium' |
  'oneLineSmall' |
  'oneLineCaption' |
  'oneLineSemibold' |
  'oneLineBoldHuge' |
  'oneLineBoldExtraLarge' |
  'oneLineBoldLarge' |
  'oneLineBoldMedium' |
  'oneLineBoldSmall';

const getHeadingStyles = (theme: AppThemeType, lineHeight: number | null) => ({
  headingDisplay: {
    ...theme.typography.weights.bold,
    fontSize: theme.typography.sizes.huge,
    lineHeight: calcLineHeight(theme.typography.sizes.huge, lineHeight ?? theme.typography.lineHeights.medium),
  },
  headingXLarge: {
    ...theme.typography.weights.bold,
    fontSize: theme.typography.sizes.XXXL,
    lineHeight: calcLineHeight(theme.typography.sizes.XXXL, lineHeight ?? theme.typography.lineHeights.medium),
  },
  headingLarge: {
    ...theme.typography.weights.bold,
    fontSize: theme.typography.sizes.XXL,
    lineHeight: calcLineHeight(theme.typography.sizes.XXL, lineHeight ?? theme.typography.lineHeights.large),
  },
  headingMedium: {
    ...theme.typography.weights.bold,
    fontSize: theme.typography.sizes.XL,
    lineHeight: calcLineHeight(theme.typography.sizes.XL, lineHeight ?? theme.typography.lineHeights.medium),
  },
  headingSmall: {
    ...theme.typography.weights.bold,
    fontSize: theme.typography.sizes.large,
    lineHeight: calcLineHeight(theme.typography.sizes.large, lineHeight ?? theme.typography.lineHeights.medium),
  },
});

const getSmallStyles = (theme: AppThemeType, lineHeight: number | null) => ({
  small: {
    ...theme.typography.weights.bold,
    fontSize: theme.typography.sizes.XS,
    lineHeight: calcLineHeight(14, lineHeight ?? theme.typography.lineHeights.medium),
  },
  smallCaps: {
    ...theme.typography.weights.bold,
    fontSize: theme.typography.sizes.XXXS,
    lineHeight: calcLineHeight(theme.typography.sizes.XXXS, lineHeight ?? theme.typography.lineHeights.medium),
    textTransform: 'uppercase' as TextStyle['textTransform'],
  },
});

const getOneLineStyles = (theme: AppThemeType, lineHeight: number | null) => ({
  oneLineHuge: {
    ...theme.typography.weights.regular,
    fontSize: theme.typography.sizes.XL,
    lineHeight: calcLineHeight(theme.typography.sizes.XL, lineHeight ?? theme.typography.lineHeights.small),
  },
  oneLineLarge: {
    ...theme.typography.weights.regular,
    fontSize: theme.typography.sizes.large,
    lineHeight: calcLineHeight(theme.typography.sizes.large, lineHeight ?? theme.typography.lineHeights.small),
  },
  oneLineMedium: {
    ...theme.typography.weights.regular,
    fontSize: theme.typography.sizes.small,
    lineHeight: calcLineHeight(theme.typography.sizes.small, lineHeight ?? theme.typography.lineHeights.small),
  },
  oneLineSmall: {
    ...theme.typography.weights.regular,
    fontSize: theme.typography.sizes.XS,
    lineHeight: calcLineHeight(theme.typography.sizes.XS, lineHeight ?? theme.typography.lineHeights.small),
  },
  oneLineCaption: {
    ...theme.typography.weights.regular,
    fontSize: theme.typography.sizes.XXS,
    lineHeight: calcLineHeight(theme.typography.sizes.XXS, lineHeight ?? theme.typography.lineHeights.small),
  },
  oneLineSemibold: {
    ...theme.typography.weights.semiBold,
    fontSize: theme.typography.sizes.medium,
    lineHeight: calcLineHeight(theme.typography.sizes.medium, lineHeight ?? theme.typography.lineHeights.small),
  },
  oneLineBoldHuge: {
    ...theme.typography.weights.bold,
    fontSize: theme.typography.sizes.XL,
    lineHeight: calcLineHeight(theme.typography.sizes.XL, lineHeight ?? theme.typography.lineHeights.small),
  },
  oneLineBoldExtraLarge: {
    ...theme.typography.weights.bold,
    fontSize: theme.typography.sizes.large,
    lineHeight: calcLineHeight(theme.typography.sizes.large, lineHeight ?? theme.typography.lineHeights.small),
  },
  oneLineBoldLarge: {
    ...theme.typography.weights.bold,
    fontSize: theme.typography.sizes.medium,
    lineHeight: calcLineHeight(theme.typography.sizes.medium, lineHeight ?? theme.typography.lineHeights.small),
  },
  oneLineBoldMedium: {
    ...theme.typography.weights.bold,
    fontSize: theme.typography.sizes.small,
    lineHeight: calcLineHeight(theme.typography.sizes.small, lineHeight ?? theme.typography.lineHeights.small),
  },
  oneLineBoldSmall: {
    ...theme.typography.weights.bold,
    fontSize: theme.typography.sizes.XS,
    lineHeight: calcLineHeight(theme.typography.sizes.XS, lineHeight ?? theme.typography.lineHeights.small),
  },
});

const getParagraphStyles = (theme: AppThemeType, lineHeight: number | null) => ({
  subtitle: {
    ...theme.typography.weights.regular,
    fontSize: theme.typography.sizes.medium,
    lineHeight: calcLineHeight(theme.typography.sizes.medium, lineHeight ?? theme.typography.lineHeights.large),
  },
  bodySmall: {
    ...theme.typography.weights.regular,
    fontSize: theme.typography.sizes.XS,
    lineHeight: calcLineHeight(theme.typography.sizes.XS, lineHeight ?? theme.typography.lineHeights.large),
  },
  caption: {
    ...theme.typography.weights.regular,
    fontSize: theme.typography.sizes.XXS,
    lineHeight: calcLineHeight(theme.typography.sizes.XXS, lineHeight ?? theme.typography.lineHeights.medium),
  },
  body: {
    ...theme.typography.weights.regular,
    fontSize: theme.typography.sizes.small,
    lineHeight: calcLineHeight(theme.typography.sizes.small, lineHeight ?? theme.typography.lineHeights.large),
  },
});

export function getVariantStyle(
  theme: AppThemeType,
  variant: TextVariantType,
  lineHeightName?: LineHeightType,
): TextStyle {
  const lineHeight = lineHeightName ? theme.typography.lineHeights[lineHeightName] : null;

  const variantStyles: Record<TextVariantType, TextStyle> = {
    // heading
    ...getHeadingStyles(theme, lineHeight),
    // small
    ...getSmallStyles(theme, lineHeight),
    // one line
    ...getOneLineStyles(theme, lineHeight),
    // paragraph
    ...getParagraphStyles(theme, lineHeight),
  };

  return variantStyles[variant];
}

export function getColorStyle(theme: AppThemeType, color: ColorPaletesType): TextStyle | null {
  return {
    color: theme.colors[color],
  };
}

export function getWeightStyle(theme: AppThemeType, weight?: FontWeightType): TextStyle | null {
  if (!weight) { return null; }

  return theme.typography.weights[weight];
}
