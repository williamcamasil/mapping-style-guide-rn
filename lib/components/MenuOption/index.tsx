import React, { ComponentType } from 'react';
import {
  StyleProp,
  StyleSheet, TouchableOpacity, View, ViewStyle,
} from 'react-native';

import { SvgProps } from 'react-native-svg';

import { Icons } from '../../assets';
import { useViewStyles } from '../../hooks/useStyles';
import { AppThemeType, withTheme } from '../../theme';
import { ColorPaletesType } from '../../tokens';
import Spacer from '../Spacer';
import Text, { TextPropTypes, TextVariantType } from '../Text';

export type MenuOptionPropsType<IconProps extends SvgProps = SvgProps> = {
  theme: AppThemeType;
  children: TextPropTypes['children'];
  onPress?: () => void;
  disabled?: boolean;
  Icon?: ComponentType<IconProps>;
  testID?: string;
  textColor?: ColorPaletesType;
  textVariant?: TextVariantType;
  iconColor?: ColorPaletesType;
  rightContent?: React.ReactNode;
  description?: string;
  descriptionColor?: ColorPaletesType;
  descriptionVariant?: TextVariantType;
  descriptionNumberOfLines?: number;
  containerStyle?: StyleProp<ViewStyle>;
};

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  containerIcon: {
    height: '100%',
    alignItems: 'flex-start',
  },
  content: {
    flex: 1,
  },
});

const MenuOption: React.FC<MenuOptionPropsType> = ({
  onPress,
  Icon,
  testID,
  textColor,
  textVariant,
  iconColor,
  rightContent,
  description,
  children,
  theme,
  disabled,
  descriptionColor,
  descriptionVariant,
  descriptionNumberOfLines,
  containerStyle,
}) => {
  const containerStyles = useViewStyles(() => [
    styles.container,
    {
      borderBottomWidth: 1,
      borderBottomColor: theme.colors.neutralGray200,
      paddingVertical: theme.spacings.sMedium,
      paddingLeft: theme.spacings.sXXS,
    },
    containerStyle,
  ], [containerStyle, theme.colors.neutralGray200, theme.spacings.sMedium, theme.spacings.sXXS]);

  const arrowContainerStyles = useViewStyles(() => [
    {
      paddingRight: theme.spacings.sXXS,
    },
  ], [theme]);

  const containerIconStyles = useViewStyles(() => [
    styles.containerIcon,
    {
      paddingRight: theme.spacings.sXXS,
      paddingTop: theme.spacings.sQuark,
    },
  ], [theme]);

  const renderIcon = () => {
    if (!Icon) return null;
    return (
      <View style={containerIconStyles} testID="icon">
        <Icon color={theme.colors[iconColor ?? 'primaryMain']} />
      </View>
    );
  };

  const renderDescription = () => {
    if (!description) return null;
    return (
      <View>
        <Spacer size={theme.spacings.sQuark} />
        <Text
          variant={descriptionVariant}
          color={descriptionColor}
          lineHeight="medium"
          numberOfLines={descriptionNumberOfLines}
        >
          {description}
        </Text>
      </View>
    );
  };

  const renderOption = () => {
    if (rightContent) return rightContent;

    return (
      <View style={arrowContainerStyles}>
        <Icons.Small.Right color={theme.colors.neutralGray500} />
      </View>
    );
  };

  return (
    <TouchableOpacity
      activeOpacity={theme.opacities.intense}
      onPress={onPress}
      disabled={!onPress || disabled}
      testID={testID}
      style={containerStyles}
    >

      {renderIcon()}
      <View style={styles.content}>
        <Text
          weight="bold"
          variant={textVariant}
          theme={theme}
          color={textColor}
        >
          {children}
        </Text>
        {renderDescription()}
      </View>
      {renderOption()}
    </TouchableOpacity>
  );
};

MenuOption.defaultProps = {
  textColor: 'neutralGray700',
  textVariant: 'body',
  descriptionColor: 'neutralGray600',
  descriptionVariant: 'caption',
};

export default withTheme(MenuOption);
