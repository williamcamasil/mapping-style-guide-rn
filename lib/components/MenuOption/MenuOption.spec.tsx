import React from 'react';
import { View } from 'react-native';

import {
  render, fireEvent,
} from '@testing-library/react-native';

import MenuOption from '.';
import { Icons } from '../../assets';
import { DefaultTheme } from '../../theme';
import Text from '../Text';

describe('MenuOption snapshots', () => {
  it('default', () => {
    const result = render(
      <MenuOption
        textColor="neutralGray700"
        description="Lorem Ipsum is simply dummy"
        Icon={Icons.Default.Email}
      >
        Minha conta
      </MenuOption>,
    ).toJSON();

    expect(result).toMatchSnapshot();
  });

  it('custom right component', () => {
    const result = render(
      <MenuOption
        textColor="neutralGray700"
        description="Lorem Ipsum is simply dummy"
        Icon={Icons.Default.Email}
        rightContent={(
          <View>
            <Text>
              Clique aqui
            </Text>
          </View>
        )}
      >
        Minha conta
      </MenuOption>,
    ).toJSON();

    expect(result).toMatchSnapshot();
  });
});

describe('MenuOption Component', () => {
  it('should render button with name title and description', async () => {
    const result = render(
      <MenuOption
        onPress={() => {}}
        textColor="neutralGray700"
        textVariant="subtitle"
        description="Lorem Ipsum is simply dummy"
      >
        Minha conta
      </MenuOption>,
    );

    await result.findByText('Lorem Ipsum is simply dummy');
    await result.findByText('Minha conta');
  });

  it('should have action in button and description', () => {
    const mockFunction = jest.fn();
    const result = render(
      <MenuOption
        onPress={mockFunction}
        textColor="neutralGray700"
        testID="button-action"
      >
        Minha conta
      </MenuOption>,
    );

    const menu = result.getByTestId('button-action');
    fireEvent.press(menu);

    expect(mockFunction).toHaveBeenCalled();
  });

  it('should render default color icon', async () => {
    const result = render(
      <MenuOption
        onPress={() => {}}
        textColor="neutralGray700"
        Icon={Icons.Default.Desktop}
      >
        Minha conta
      </MenuOption>,
    );

    expect(result.getByTestId('icon')).toBeTruthy();
  });

  it('should render description with prop numberOfLines to equal 1', async () => {
    const result = render(
      <MenuOption
        onPress={() => {}}
        textColor="neutralGray700"
        Icon={Icons.Default.Desktop}
        descriptionNumberOfLines={1}
        description="Lorem Ipsum is simply dummy"
      >
        Minha conta
      </MenuOption>,
    );

    const text = result.getByText('Lorem Ipsum is simply dummy');

    expect(text.props.numberOfLines).toBe(1);
  });

  it('should render icon and arrow', async () => {
    const result = render(
      <MenuOption
        onPress={() => {}}
        textColor="neutralGray700"
        Icon={Icons.Default.Desktop}
        iconColor="neutralBlack"
        rightContent={<Icons.Small.Right color={DefaultTheme.colors.neutralGray500} testID="arrow-menu" />}
      >
        Minha conta
      </MenuOption>,
    );

    expect(result.getByTestId('icon')).toBeTruthy();
    expect(result.getByTestId('arrow-menu')).toBeTruthy();
  });
});
