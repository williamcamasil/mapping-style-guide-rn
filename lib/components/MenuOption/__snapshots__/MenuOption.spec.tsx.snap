// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuOption snapshots custom right component 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": true,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "borderBottomColor": "#F0F0F2",
      "borderBottomWidth": 1,
      "flexDirection": "row",
      "opacity": 1,
      "paddingLeft": 12,
      "paddingVertical": 24,
    }
  }
>
  <View
    style={
      {
        "alignItems": "flex-start",
        "height": "100%",
        "paddingRight": 12,
        "paddingTop": 4,
      }
    }
    testID="icon"
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="24"
      bbWidth="24"
      fill="none"
      focusable={false}
      height={24}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "flex": 0,
            "height": 24,
            "width": 24,
          },
        ]
      }
      vbHeight={24}
      vbWidth={24}
      width={24}
    >
      <RNSVGGroup
        fill={null}
        propList={
          [
            "fill",
          ]
        }
      >
        <RNSVGPath
          d="M17 21h-5v0a9 9 0 1 1 9-9v1.5a2.5 2.5 0 0 1-5 0V12a4 4 0 0 0-4-4v0a4 4 0 1 0 4 4"
          fill={
            {
              "payload": 4278190080,
              "type": 0,
            }
          }
          propList={
            [
              "stroke",
              "strokeWidth",
              "strokeLinecap",
              "strokeLinejoin",
            ]
          }
          stroke={
            {
              "payload": 4278268296,
              "type": 0,
            }
          }
          strokeLinecap={1}
          strokeLinejoin={1}
          strokeWidth="1.8"
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
  <View
    style={
      {
        "flex": 1,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        {
          "color": "#3E4149",
          "flexShrink": 1,
          "fontFamily": "Averta CY",
          "fontSize": 16,
          "fontWeight": "bold",
          "letterSpacing": -0.02,
          "lineHeight": 22,
        }
      }
    >
      Minha conta
    </Text>
    <View>
      <Text
        style={
          {
            "height": 4,
            "width": 4,
          }
        }
      />
      <Text
        allowFontScaling={false}
        style={
          {
            "color": "#676D79",
            "flexShrink": 1,
            "fontFamily": "Averta CY",
            "fontSize": 13,
            "fontWeight": "normal",
            "letterSpacing": -0.02,
            "lineHeight": 16,
          }
        }
      >
        Lorem Ipsum is simply dummy
      </Text>
    </View>
  </View>
  <View>
    <Text
      allowFontScaling={false}
      style={
        {
          "color": "#676D79",
          "flexShrink": 1,
          "fontFamily": "Averta CY",
          "fontSize": 16,
          "fontWeight": "normal",
          "letterSpacing": -0.02,
          "lineHeight": 22,
        }
      }
    >
      Clique aqui
    </Text>
  </View>
</View>
`;

exports[`MenuOption snapshots default 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": true,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "borderBottomColor": "#F0F0F2",
      "borderBottomWidth": 1,
      "flexDirection": "row",
      "opacity": 1,
      "paddingLeft": 12,
      "paddingVertical": 24,
    }
  }
>
  <View
    style={
      {
        "alignItems": "flex-start",
        "height": "100%",
        "paddingRight": 12,
        "paddingTop": 4,
      }
    }
    testID="icon"
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="24"
      bbWidth="24"
      fill="none"
      focusable={false}
      height={24}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "flex": 0,
            "height": 24,
            "width": 24,
          },
        ]
      }
      vbHeight={24}
      vbWidth={24}
      width={24}
    >
      <RNSVGGroup
        fill={null}
        propList={
          [
            "fill",
          ]
        }
      >
        <RNSVGPath
          d="M17 21h-5v0a9 9 0 1 1 9-9v1.5a2.5 2.5 0 0 1-5 0V12a4 4 0 0 0-4-4v0a4 4 0 1 0 4 4"
          fill={
            {
              "payload": 4278190080,
              "type": 0,
            }
          }
          propList={
            [
              "stroke",
              "strokeWidth",
              "strokeLinecap",
              "strokeLinejoin",
            ]
          }
          stroke={
            {
              "payload": 4278268296,
              "type": 0,
            }
          }
          strokeLinecap={1}
          strokeLinejoin={1}
          strokeWidth="1.8"
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
  <View
    style={
      {
        "flex": 1,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        {
          "color": "#3E4149",
          "flexShrink": 1,
          "fontFamily": "Averta CY",
          "fontSize": 16,
          "fontWeight": "bold",
          "letterSpacing": -0.02,
          "lineHeight": 22,
        }
      }
    >
      Minha conta
    </Text>
    <View>
      <Text
        style={
          {
            "height": 4,
            "width": 4,
          }
        }
      />
      <Text
        allowFontScaling={false}
        style={
          {
            "color": "#676D79",
            "flexShrink": 1,
            "fontFamily": "Averta CY",
            "fontSize": 13,
            "fontWeight": "normal",
            "letterSpacing": -0.02,
            "lineHeight": 16,
          }
        }
      >
        Lorem Ipsum is simply dummy
      </Text>
    </View>
  </View>
  <View
    style={
      {
        "paddingRight": 12,
      }
    }
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="16"
      bbWidth="16"
      fill="none"
      focusable={false}
      height={16}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "flex": 0,
            "height": 16,
            "width": 16,
          },
        ]
      }
      vbHeight={16}
      vbWidth={16}
      width={16}
    >
      <RNSVGGroup
        fill={null}
        propList={
          [
            "fill",
          ]
        }
      >
        <RNSVGPath
          d="m10 12 4-4-4-4"
          fill={
            {
              "payload": 4278190080,
              "type": 0,
            }
          }
          propList={
            [
              "stroke",
              "strokeWidth",
              "strokeLinecap",
              "strokeLinejoin",
            ]
          }
          stroke={
            {
              "payload": 4288980911,
              "type": 0,
            }
          }
          strokeLinecap={1}
          strokeLinejoin={1}
          strokeWidth="2"
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
</View>
`;
